bootloader/grub.go

```diff
@@ -496,21 +496,69 @@ func staticCommandLineForGrubAssetEdition(asset string, edition uint) string {
 	return string(cmdline)
 }
 
+type taggedPath struct {
+	tag  string
+	path string
+}
+
+func (t taggedPath) Id() string {
+	basename := filepath.Base(t.path)
+	if t.tag == "" {
+		return basename
+	}
+	return fmt.Sprintf("%s:%s", t.tag, basename)
+}
+
 // grubBootAssetPath contains the paths for assets in the boot chain.
 type grubBootAssetPath struct {
-	shimBinary string
-	grubBinary string
+	defaultShimBinary taggedPath
+	defaultGrubBinary taggedPath
+	fallbackBinary    taggedPath
+	shimBinary        taggedPath
+	grubBinary        taggedPath
 }
 
 // grubBootAssetsForArch contains the paths for assets for different
 // architectures in a map
 var grubBootAssetsForArch = map[string]grubBootAssetPath{
 	"amd64": {
-		shimBinary: filepath.Join("EFI/boot/", "bootx64.efi"),
-		grubBinary: filepath.Join("EFI/boot/", "grubx64.efi")},
+		defaultShimBinary: taggedPath{
+			path: filepath.Join("EFI/boot/", "bootx64.efi"),
+		},
+		defaultGrubBinary: taggedPath{
+			path: filepath.Join("EFI/boot/", "grubx64.efi"),
+		},
+		fallbackBinary: taggedPath{
+			path: filepath.Join("EFI/boot/", "fbx64.efi"),
+		},
```

Comment:
Yes, I should may comment on it. The idea about using "asset names" instead of the path is because current implementation uses base names. So for backward compliance, we need to use no tag for all the paths that were used before. I will comment it.

Reply:
Ah now I understand. fb was not an old one. I can put boot: for this one.